agent_app_sample_code/01_data_pipeline.py

```diff
@@ -0,0 +1,580 @@
+# MAGIC
+# MAGIC 🚫✏️ Only modify if you need additional packages in your code changes to the document parsing or chunking logic.
+
+# COMMAND ----------
+
+# Versions of Databricks code are not locked since Databricks ensures changes are backwards compatible.
+# Versions of open source packages are locked since package authors often make backwards compatible changes
+%pip install -qqqq -U \
+  -r requirements.txt `# Packages shared across all notebooks` \
+  pymupdf4llm==0.0.5 pymupdf==1.24.5 `# PDF parsing` \
+  markdownify==0.12.1  `# HTML parsing` \
+  pypandoc_binary==1.13  `# DOCX parsing` \
+  transformers==4.41.1 torch==2.3.0 tiktoken==0.7.0 langchain-text-splitters==0.2.0. `# For get_recursive_character_text_splitter()`
+
+# Restart to load the packages into the Python environment
+dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC
+# MAGIC ## 1️⃣ 📂 Data source & destination configuration
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC #### 🚫✏️ (Optional) Change Agent's shared storage location configuration
+# MAGIC
+# MAGIC **If you configured `00_shared_config`, just run this cell as-is.**
+# MAGIC
+# MAGIC From the shared configuration, we use:
+# MAGIC * The UC catalog & schema as the default location for output tables/indexs
+# MAGIC * The MLflow experiment for tracking pipeline runs
+# MAGIC
+# MAGIC *These values can be set here if you want to use this notebook independently.*
+
+# COMMAND ----------
+
+from cookbook_utils.cookbook_config import AgentCookbookConfig
+import mlflow
+
+# Load the shared configuration
+cookbook_shared_config = AgentCookbookConfig.from_yaml_file('./configs/cookbook_config.yaml')
+
+# Print configuration 
+cookbook_shared_config.pretty_print()
+
+# Set the MLflow Experiment that is used to track metadata about each run of this Data Pipeline.
+experiment_info = mlflow.set_experiment(cookbook_shared_config.mlflow_experiment_name)
+
+# COMMAND ----------
+
+# MAGIC %md #### 📂 ✅✏️ Configure the data pipeline's source location.
+# MAGIC
+# MAGIC Choose a [Unity Catalog Volume](https://docs.databricks.com/en/volumes/index.html) containing PDF, HTML, etc documents to be parsed/chunked/embedded.
+# MAGIC
+# MAGIC - `uc_catalog_name`: Name of the Unity Catalog.
+# MAGIC - `uc_schema_name`: Name of the Unity Catalog schema.
+# MAGIC - `uc_volume_name`: Name of the Unity Catalog volume. 
+# MAGIC
+# MAGIC Running this cell with validate that the UC Volume exists, trying to create it if not.
+# MAGIC
+
+# COMMAND ----------
+
+from datapipeline_utils.data_pipeline_config import UnstructuredDataPipelineSourceConfig
+
+# Default is a Volume called `{uc_asset_prefix}_source_docs` in the configured UC catalog/schema
+source_config = UnstructuredDataPipelineSourceConfig(
+    uc_catalog_name=cookbook_shared_config.uc_catalog_name,
+    uc_schema_name=cookbook_shared_config.uc_schema_name,
+    uc_volume_name=f"{cookbook_shared_config.uc_asset_prefix}_source_docs"
```

Comment:
My source docs are in a different catalog - do I need to move them? I will try it referencing the other catalog and see what breaks

Reply:
The implementation of this config dataclass would need to be changed if I want to use data from another catalog without creating a new volume. This would require data copying

Reply:
When I tried a configuration from scratch I got something like:

```
app_name: alphaledger
data_sources: 
  sec_10k:
    catalog: field_ai_examples
    schema: alphaledger
    source_type: volume
    path: '/pdf'
    format: pdf
    glob: '*10k.pdf'
  sec_10q:
    catalog: field_ai_examples
    schema: alphaledger
    source_type: volume
    path: '/pdf'
    format: pdf
    glob: '*10k.pdf'
  marketdata:
    catalog: dev
    schema: alphaledger
    source_type: delta

data_pipeline:
  stages:
    - name: ingest_sec
      sources:
        - sec_10k
        - sec_10q
```

This makes it easier to define semantics of data in the Volume, but makes the ingest job more complex